crates/uv-requirements-txt/src/lib.rs

```diff
@@ -879,6 +896,47 @@ fn parse_value<'a, T>(
     }
 }
 
+/// Given a quoted string (like `"foo"`), strip the quotes, taking into account escaped inner quotes
+/// (e.g., given `"foo \"bar\""`, return `foo "bar"`.
+///
+/// If the quotes are unbalanced, return `None`.
+fn strip_quotes(s: &str) -> Option<String> {
+    let mut chars = s.chars();
+
+    // Ensure the string starts and ends with the same quote type
+    let quote = chars.next()?;
+    if quote != '"' && quote != '\'' {
+        return None;
+    }
+    if !s.ends_with(quote) || s.len() < 2 {
+        return None;
+    }
+
+    let mut result = String::with_capacity(s.len() - 2);
+    let mut escaped = false;
+
+    for c in chars.take(s.len() - 2) {
+        if escaped {
+            result.push(c);
+            escaped = false;
+        } else if c == '\\' {
+            escaped = true;
```

Comment:
Escaping only works for double quotes. Single quotes preserve their content as-is. So one cannot include single quote inside single quotes. 

There are a lot more weird rules to POSIX parsing, which is what pip relies on. It might be better to offload this to some shell parser. For example even in double quotes if the forward slash is followed by non-escape character, the forward slash and character both will be preserved.

Just some examples:
- `-e "foo""bar"` is valid and is equivalent to `-c foobar`
- `-e 'foo'\''bar'` is equivalent to `-c "foo'bar"`
- `-e "\ab"` preserves the forward slash also

[reference](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#:~:text=a%20token%20separator.-,2.2.2%20Single%2DQuotes,-Enclosing%20characters%20in)